### PR TITLE
coverage.yml: fail_ci_if_error = true

### PR DESCRIPTION
**Pull request by wdconinc**
Failing coverage CI runs are marked as success, e.g. in https://github.com/spack/spack/actions/runs/11943108426/job/33292879311?pr=47706#step:9:266 for #47706. This can lead to merging PRs that are actually breaking things.

This PR adds the `fail_ci_if_error` action argument and sets it to `true` (default = false).

Other considerations:
- Should this be restricted to `true` only when `github.repository == "spack/spack"` to avoid breaking workflows on forks? Edit: I don't think this is needed.

### CODE REVIEW

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks for catching this @wdconinc!

**Comment by tgamblin** [comment]
FWIW, we explicitly made this NOT fail on error because we didn't want to hold up CI if codecov upload fails.  Our coverage checks are more reliable now, so maybe it's ok to do this in general, but even right now this is going to break all our CI until we fix #47953.  Do we really want that?  I don't think so.

You could instead add this option to the branch for #47953 so that things fail there while we get coverage working, *then* we could decide if we want to keep it.  But as flaky as `codecov` has been I worry about making it mandatory.

@adamjstewart FYI

**Comment by wdconinc** [comment]
> but even right now this is going to break all our CI until we fix #47953.

This won't fail *unless* we also merge #47953. Current workflows, from spack org and from forks, are not going to fail to upload (as far as I understand).

**Comment by tgamblin** [comment]
> This won't fail unless we also merge https://github.com/spack/spack/pull/47953. Current workflows, from spack org and from forks, are not going to fail to upload (as far as I understand).

This sounds ok to me, although it also means that if coverage uploads fail for reasons unrelated to PRs, then PRs are going to fail.  We've historically left that up to maintainers. If we want to make it mandatory and force PRs to wait (if something fails), I'm ok with it.

**Comment by alalazo** [comment]
Need to check if this is a ci bug (we forgot to use GITHUB_TOKEN somewhere) revealed by this PR, or if it is just due to being more strict on upload failures: https://github.com/spack/spack/actions/runs/12211650410/job/34069701318

**Pull request merged**
